lib/posthog/feature_flags.rb

```diff
@@ -46,30 +48,22 @@ def load_feature_flags(force_reload = false)
       end
     end
 
-    def get_feature_variants(distinct_id, groups={}, person_properties={}, group_properties={})
-      decide_data = get_decide(distinct_id, groups, person_properties, group_properties)
+    def get_feature_variants(distinct_id, groups={}, person_properties={}, group_properties={}, raise_on_error=false)
+      # TODO: Convert to options hash for easier argument passing
+      decide_data = get_all_flags_and_payloads(distinct_id, groups, person_properties, group_properties, false, raise_on_error)
       if !decide_data.key?(:featureFlags)
-        logger.error "Missing feature flags key: #{decide_data.to_json}"
+        logger.debug "Missing feature flags key: #{decide_data.to_json}"
+        return {}
```

Comment:
```suggestion
        {}
```

Reply:
Hmmm, I _thought_ the no return applied only at the end of the block, can I also do this in the middle of the flow?

(this is confusing me, sorry 😅 ). Just for my understanding, what if I had something else on line 59.5 , will this still return early?

Reply:
The last executed and thus the last returned line will either be the last line of the `if` block or the last and only line of the `else` block

Reply:
oh, that's a better way to think about it, 'the last executed line'. So if at the end of the else block, I added another line that was outside the else block, that would've been returned; but since there is no such line, the last executed one is the last line of the if block. Nice, thanks 👍